source/common/memory/stats.cc

```diff
@@ -4,7 +4,53 @@
 
 #include "common/common/logger.h"
 
-#ifdef TCMALLOC
+#if defined(TCMALLOC)
+
+#include "tcmalloc/malloc_extension.h"
+
+namespace Envoy {
+namespace Memory {
+
+uint64_t Stats::totalCurrentlyAllocated() {
+  return tcmalloc::MallocExtension::GetNumericProperty("generic.current_allocated_bytes")
```

Comment:
maybe for deterministic memory checks we can just use generic.current_allocated_bytes and not include tcmalloc.cpu_free?

We can make 2 APIs if that helps.

I'm also confused by the github UI; it is not showing me the prior impl of this method.

Reply:
This part (from `#if` to `#elif`) is new for tcmalloc. The previous impl is below.

Reply:
> maybe for deterministic memory checks we can just use generic.current_allocated_bytes and not include tcmalloc.cpu_free?

Right, my bad. It shouldn't be included. I'll drop it. Though after the last merge all tests passed for  https://github.com/envoyproxy/envoy/pull/13251/commits/a6d4c424ce5c5c2c01c37557632e6481f028ac4c. But the version without tcmalloc.cpu_free passes the release tests too on my local host. Let's see...

> I'm also confused by the github UI; it is not showing me the prior impl of this method.

The old version is still there, just wrapped in `#if defined(GPERFTOOLS_TCMALLOC)`. It's needed for non-x86 builds.